Tests/TensorFlowTests/TensorAutoDiffTests.swift

```diff
@@ -46,7 +46,7 @@ final class TensorAutoDiffTests: XCTestCase {
         func negate<T : TensorFlowFloatingPoint>(_ x: Tensor<T>) -> Tensor<T> {
             return (1 - x).sum()
         }
-        XCTAssertEqual(gradient(at: Tensor([0.1, 0.2, 0.3]), in: negate), Tensor(-1))
+        XCTAssertEqual(gradient(at: Tensor([0.1, 0.2, 0.3]), in: negate), Tensor([-1, -1, -1]))
```

Comment:
Afaik, broadcasting is at works here.

Reply:
The TODO message originally written in `==` is:

> // TODO: This is not correct due to broadcasting.

It make sense to disable broadcastring. So I disabled it in this PR.

Reply:
My first PR comment didn't described the change enough.
I updated it.

Reply:
In this specific case, since its a 1-D broadcast, I thought it was okay. The case you mentioned in the original comment though makes sense.